File: javascript-scripts/json-scrubber.js
const fs = require('fs');
const path = require('path');

const inputFilename = path.join('data', process.env.INPUT_FILE || 'data/input.json');
const outputFilename = path.join('data', process.env.OUTPUT_FILE || 'data/output.json');
const keepIfMatched = process.env.KEEP_IF_MATCHED ? process.env.KEEP_IF_MATCHED.split(',') : [];

// Read data from the input file
const inputData = fs.readFileSync(inputFilename, 'utf8');
const data = JSON.parse(inputData);

// Process the data
data._source.metadata = data._source.metadata.filter(item => !item.name || keepIfMatched.includes(item.name));

// Export the modified data to an output file
fs.writeFileSync(outputFilename, JSON.stringify(data, null, 4), 'utf8');

console.log(`Modified data exported to ${outputFilename}`);

File: python-scripts/json-scrubber.py
import os
import json

# Read data from a file
input_filename =  "data/" + os.environ.get("INPUT_FILE", "data/input.json")
output_filename = "data/" + os.environ.get("OUTPUT_FILE", "data/output.json")
keepIfMatched = os.environ.get("KEEP_IF_MATCHED", "").split(",")

# Read data from the input file
with open(input_filename, "r") as input_file:
    data = json.load(input_file)

# Process the data
data["_source"]["metadata"] = [
    item for item in data["_source"]["metadata"]
    if "name" not in item or item["name"] in keepIfMatched
]

# Export the modified data to an output file
with open(output_filename, "w") as output_file:
    json.dump(data, output_file, indent=4)

print("Modified data exported to", output_filename)